presto-tpcds/src/test/java/com/facebook/presto/sql/planner/optimizations/TestTpcdsCostBasedPlan.java

```diff
@@ -0,0 +1,107 @@
+import com.google.common.collect.ImmutableMap;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
+import static com.facebook.presto.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static java.lang.String.format;
+
+/**
+ * This class tests cost-based optimization rules related to joins. It contains unmodified TPCDS queries.
+ * This class is using TPCDS connector configured in way to mock Hive connector with unpartitioned TPCDS tables.
+ */
+public class TestTpcdsCostBasedPlan
+        extends BaseCostBasedPlanTest
+{
+    /*
+     * CAUTION: The expected plans here are not necessarily optimal yet. Their role is to prevent
+     * inadvertent regressions. A conscious improvement to the planner may require changing some
+     * of the expected plans, but any such change should be verified on an actual cluster with
+     * large amount of data.
+     */
+
+    public TestTpcdsCostBasedPlan()
+    {
+        super(() -> {
+            String catalog = "local";
+            Session.SessionBuilder sessionBuilder = testSessionBuilder()
+                    .setCatalog(catalog)
+                    .setSchema("sf3000.0")
+                    .setSystemProperty("task_concurrency", "1") // these tests don't handle exchanges from local parallel
+                    .setSystemProperty(JOIN_REORDERING_STRATEGY, JoinReorderingStrategy.AUTOMATIC.name())
+                    .setSystemProperty(JOIN_DISTRIBUTION_TYPE, JoinDistributionType.AUTOMATIC.name());
+
+            LocalQueryRunner queryRunner = LocalQueryRunner.queryRunnerWithFakeNodeCountForStats(sessionBuilder.build(), 8);
+            queryRunner.createCatalog(
+                    catalog,
+                    new TpcdsConnectorFactory(1),
+                    ImmutableMap.of());
+            return queryRunner;
+        });
+    }
+
+    @Override
+    protected Stream<Query> getQueries()
+    {
+        return IntStream.range(1, 100)
+                .boxed()
+                .flatMap(i -> {
+                    String queryId = format("q%02d", i);
+                    if (i == 14 || i == 23 || i == 24 || i == 39) {
+                        return Stream.of(queryId + "_1", queryId + "_2");
+                    }
+                    return Stream.of(queryId);
+                })
+                .map(queryId -> new Query(queryId, tpcdsQuery(queryId)));
+    }
+
+    private static Path tpcdsQuery(String queryId)
+    {
+        return Paths.get(format("./presto-benchto-benchmarks/src/main/resources/sql/presto/tpcds/%s.sql", queryId));
+    }
+
+    @Override
+    protected Path getExpectedJoinOrderingFile(String queryId)
+    {
+        return Paths.get(format("./presto-tpcds/src/test/resources/tpcds-join-ordering/%s.txt", queryId));
```

Comment:
Ditto about paths

Reply:
It is needed to expected query plans generation.